vendor/bat-native-ledger/src/bat/ledger/internal/grants.cc

```diff
@@ -124,29 +148,38 @@ void Grants::GetGrantsCallback(
 }
 
 void Grants::SetGrant(const std::string& captchaResponse,
-                      const std::string& promotionId) {
-  if (promotionId.empty()) {
+                      const std::string& promotionId,
+                      const std::string& safetynet_token) {
+  if (promotionId.empty() && safetynet_token.empty()) {
     braveledger_bat_helper::GRANT properties;
     ledger_->OnGrantFinish(ledger::Result::LEDGER_ERROR, properties);
     return;
   }
 
   std::string keys[2] = {"promotionId", "captchaResponse"};
   std::string values[2] = {promotionId, captchaResponse};
-  std::string payload = braveledger_bat_helper::stringify(keys, values, 2);
+  std::string payload = braveledger_bat_helper::stringify(keys, values,
+      safetynet_token.empty() ? 2 : 1);
```

Comment:
meaning of `2` and `1`, can we please enumerate these values to give context

Reply:
NONCRITICAL
Covered in: brave/brave-browser#6235 (24)